samples/tab-nested-auth/nodejs/src/components/Msal.js

```diff
@@ -0,0 +1,143 @@
+// <copyright file="MeetingTranscriptRecording.jsx" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// </copyright>
+
+import React, { useState, useEffect } from 'react';
+import { Button } from '@fluentui/react-components';
+import { createNestablePublicClientApplication } from "@azure/msal-browser";
+import { app, nestedAppAuth } from "@microsoft/teams-js";
+
+const Msal = () => {
+
+    const [meData, setMeData] = useState(null); // State to store user data
+    const [isLoggedIn, setIsLoggedIn] = useState(false); // State to track login status
+
+    let pca = undefined; // Public client application instance
+
+    // MSAL configuration
+    const msalConfig = {
+        auth: {
+            clientId: "{{clientId}}",
+            authority: "https://login.microsoftonline.com/common"
+        }
+    };
+
+    // Function to initialize the public client application
+    function initializePublicClient() {
+        console.log("Starting initializePublicClient");
+        return createNestablePublicClientApplication(msalConfig).then(
+            (result) => {
+                console.log("Client app created");
+                pca = result;
+                return pca;
+            }
+        );
+    }
+
+    // Function to get the active account
+    const getActiveAccount = async () => {
+        console.log("Starting getActiveAccount");
+        let activeAccount = null;
+        try {
+            console.log("Getting active account");
+            activeAccount = pca.getActiveAccount();
+        } catch (error) {
+            console.log(error);
+        }
+        if (!activeAccount) {
+            console.log("No active account, trying login popup");
+            try {
+                const context = await app.getContext();
+                const accountFilter = {
+                    tenantId: context.user?.tenant?.id,
+                    homeAccountId: context.user?.id,
+                    loginHint: (await app.getContext()).user?.loginHint
```

Comment:
Is there a reason we're querying the context again here instead of using the instance we already have?

Reply:
Done